domain/model/service/service.go

```diff
@@ -542,3 +589,45 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// WatchActivatedModels returns a watcher that reports all activated models in the controller.
+// The watcher outputs events for when an activated model receives an update.
+// Deletion of activated models are not reported.
+func (s *WatchableService) WatchActivatedModels(ctx context.Context) (watcher.StringsWatcher, error) {
+	mapper := func(ctx context.Context, db database.TxnRunner,
+		changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+		uuidToChangeEventMap := make(map[string]changestream.ChangeEvent)
+		modelUUIDs := make([]string, len(changes))
+
+		for i, change := range changes {
+			modelUUID := change.Changed()
+			modelUUIDs[i] = modelUUID
+			uuidToChangeEventMap[modelUUID] = change
+		}
+
+		// Retrieve all activate status of all models with associated uuids
+		activatedModelUUIDs, err := s.st.GetActivatedModelUUIDs(ctx, modelUUIDs)
+
+		// There should be no errors returned if there are no activated models found.
+		if err != nil {
+			s.logger.Errorf(ctx, "failed to activated model UUIDs: %v\n", err)
```

Comment:
Do we need this log?

Reply:
hi @ycliuhw , would it be better if i change to
			s.logger.Errorf(ctx, "failed to retrieve activated model UUIDs for UUIDs %v: %v", modelUUIDs, err)?
			
In the event there is a db error while trying to retrieve the activated model UUIDs and we could verify which models activated updates are yet to be tracked?